client/state/data-layer/wpcom/read/streams/test/index.js

```diff
@@ -0,0 +1,300 @@
+				onSuccess: action,
+				onFailure: action,
+			} ) );
+		} );
+
+		it( 'should have called next with the original action', () => {
+			expect( next ).to.have.been.calledWith( action );
+		} );
+
+		it( 'should ignore a second action with the same params', () => {
+			next.reset();
+			requestPage( { dispatch }, action, next );
+			expect( dispatch ).to.have.been.calledOnce;
+			expect( next ).to.have.been.calledWith( action );
+		} );
+	} );
+
+	describe( 'handlePage', () => {
+		const next = spy();
+		const dispatch = spy();
+		const data = deepfreeze( {
+			posts: []
+		} );
+
+		before( () => {
+			handlePage( { dispatch }, action, next, data );
+		} );
+
+		it( 'should dispatch receivePage', () => {
+			expect( dispatch ).to.have.been.calledWith( receivePage( action.streamId, action.query, data.posts ) );
+		} );
+
+		it( 'should swallow the original action', () => {
+			expect( next ).to.not.have.been.called;
+		} );
+	} );
+
+	describe( 'handleError', () => {
+		const next = spy();
+		const dispatch = spy();
+		const error = { error: true };
+
+		before( () => {
+			handleError( { dispatch }, action, next, error );
+		} );
+
+		it( 'should dispatch a notice about the error', () => {
+			const notice = errorNotice( 'Could not fetch the next page of results' );
+			delete notice.notice.noticeId;
+			expect( dispatch ).to.have.been.calledWithMatch( notice );
+		} );
+	} );
+
+	describe( 'duplicate requests', () => {
+		afterEach( () => {
+			clearInflight();
+		} );
+
+		it( 'should ignore duplicate requests for the same query', () => {
+			const dispatch = spy();
+			const next = spy();
+			requestPage( { dispatch }, action, next );
+			requestPage( { dispatch }, action, next );
+			expect( dispatch ).has.been.calledOnce;
+			expect( next ).has.been.calledTwice;
+		} );
+
+		it( 'should allow more requests once the current one happily completes', () => {
+			const dispatch = spy();
+			const next = spy();
+			requestPage( { dispatch }, action, next );
+			handlePage( { dispatch }, action, next, { posts: [] } );
+			dispatch.reset();
+			next.reset();
+			requestPage( { dispatch }, action, next );
+			expect( dispatch ).to.have.been.calledWith( http( {
+				method: 'GET',
+				path: '/read/following',
+				apiVersion: '1.2',
+				query: action.query,
+				onSuccess: action,
+				onFailure: action,
+			} ) );
+		} );
+
+		it( 'should allow more requests even if the current one completes in error', () => {
+			const dispatch = spy();
+			const next = spy();
+			requestPage( { dispatch }, action, next );
+			handleError( { dispatch }, action, next, { error: true } );
+			dispatch.reset();
+			next.reset();
+			requestPage( { dispatch }, action, next );
+			expect( dispatch ).to.have.been.calledWith( http( {
+				method: 'GET',
+				path: '/read/following',
+				apiVersion: '1.2',
+				query: action.query,
+				onSuccess: action,
+				onFailure: action
+			} ) );
+		} );
+	} );
+
+	describe( 'transformResponse', () => {
```

Comment:
maybe add in a happy case here:
it ( 'should return the response array when data is truthy' )